truffle/com.oracle.truffle.sl/src/com/oracle/truffle/sl/nodes/SLStatementNode.java

```diff
@@ -88,6 +83,19 @@ public SLStatementNode getNonWrapperNode() {
         return this;
     }
 
+    public void setDebugHalt(boolean isDebugHalt) {
+        this.isDebugHalt = isDebugHalt;
+    }
+
+    @Override
+    protected boolean isTaggedWith(String tag) {
+        switch (tag) {
+            case Debugger.HALT_TAG:
```

Comment:
I would need to check for the parent beeing a wrapper. Also isTaggedWith is defined as unchanged after AST creation, its hard to gurantee that if the parent is changing. Also its not true that each statement of a block is tagged halt in SL. For conditionals you want to step before the condition and not the full if expression. So the boolean solution is a lot easier to fine-tune.